modules/nlightning/src/Bolts/BOLT11/InvoiceBridge.cs

```diff
@@ -2,26 +2,76 @@
 
 namespace NLightning.CppBridge.Bolts.BOLT11;
 
+using System.Text;
 using NLightning.Bolts.BOLT11;
 public static class InvoiceBridge
 {
     [UnmanagedCallersOnly(EntryPoint = "DecodeInvoice")]
-    public static bool DecodeInvoice(IntPtr invoiceStringPtr)
+    public static IntPtr DecodeInvoice(IntPtr invoiceStringPtr)
     {
         try
         {
             string? invoiceString = Marshal.PtrToStringUTF8(invoiceStringPtr);
             if (string.IsNullOrEmpty(invoiceString))
             {
-                return false;
+                return IntPtr.Zero;
             }
 
-            _ = Invoice.Decode(invoiceString);
-            return true;
+            string result = string.Empty;
```

Comment:
Nit:  I'm not a C# expert, but I think StringBuilder is the more efficient way to do this.